nvflare/app_opt/xgboost/histogram_based_v2/sender.py

```diff
@@ -11,78 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from abc import ABC, abstractmethod
 
-from nvflare.apis.shareable import ReturnCode, Shareable
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.shareable import Shareable
 from nvflare.apis.signal import Signal
-from nvflare.app_opt.xgboost.histogram_based_v2.defs import Constant
 from nvflare.fuel.f3.cellnet.fqcn import FQCN
-from nvflare.fuel.utils.obj_utils import get_logger
 
 
-class Sender:
-    """
-    A Sender is used to send XGB requests from the client to the server and wait for reply.
-    TBD: currently the sender simply sends the request with an aux message. It will be enhanced to be more
-    reliable in dealing with unstable network.
-    """
+class Sender(FLComponent, ABC):
+    """An abstract interface to send request to XGB server"""
 
-    def __init__(self, engine, timeout):
-        """Constructor
+    @abstractmethod
+    def send_to_server(self, engine, topic: str, req: Shareable, abort_signal: Signal, timeout: float):
```

Comment:
Can we change the signature and move abort_signal to be the last arg? This is how we do it everywhere else. Let's be consistent.

Reply:
Changed. The timeout was put at the last position because it had a default value in SimpleSender. It's not a good idea to have a default value in extended class so it's removed.